api/tools/src/main/java/org/pmiops/workbench/tools/ApplicationRunListener.java

```diff
@@ -0,0 +1,71 @@
+package org.pmiops.workbench.tools;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.SpringApplicationRunListener;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.support.SimpleThreadScope;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.web.context.WebApplicationContext;
+
+/**
+ * A custom application listener to allow us to mutate the application context before any
+ * command-line tools are run.
+ *
+ * <p>This class currently serves a single purpose, which is to create a fake "request" bean scope
+ * to allow command-line tools to load RW beans that are request-scoped for use in the main webapp.
+ *
+ * <p>See https://stackoverflow.com/a/28275111 which is the pattern we roughly followed here. This
+ * class is referenced from resources/META-INF/spring.factories, which causes this class to be
+ * loaded before the Spring context is initialized.
+ */
+public class ApplicationRunListener implements SpringApplicationRunListener {
```

Comment:
nit: maybe name this `ToolApplicationRunListener` so it's more obvious when it gets pulled in.

Reply:
Good idea – I renamed to CommandLineToolApplicationRunListener for (1) consistency with CommandLineToolConfig and (2) ApproachingPeakJavaCamelCase